models/new_model

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+model_name=$1
+location_mod=$2
+
+if [[ ("$1" = "help") || ("$1" = "--help") || ($# -eq 0) ]]; then
+  echo "Usage: $0 [model_name] [location_mod]"
+  exit 0
+elif [[ -d "$model_name" ]]; then
+  echo "$model_name already exists! Please try a different model name."
+  exit 1
+elif [[ $# -eq 1 ]]; then
+  echo "Please enter a proper location! (Hint: check DART/assimilation_code/location for location modules avaliable.)"
+  exit 1
+elif [[ ! -d "../assimilation_code/location/$location_mod" ]]; then
+  echo "$location_mod is not a proper location module! Please try a different location name."
+  exit 1
+else
+  cp -R template "$model_name"
+  rm -r "$model_name"/shell_scripts "$model_name"/model_mod.nml "$model_name/readme.rst"
+  mv "$model_name/new_model.rst" "$model_name/readme.rst"
+  sed -i '' "s/template_model/$model_name/" "$model_name/readme.rst"
+  sed -i '' -e  "s/template_model/$model_name/" -e "s/template_location/$location_mod/" "$model_name/work/quickbuild.sh"
+  case $location_mod in
+    threed | threed_cartesian | threed_sphere) # threed does not compile, but is a good framework
+      rm -r "$model_name"/oned_model_mod.f90 "$model_name"/work/oned_input.nml
```

Comment:
-r (recursive) not needed here, since you are not removing directories. 

Some systems will have rm aliased to rm -i. If you want to a command, e.g.  rm and not any aliases, add \ to the start of the command:

```
\rm
```    

Also, since users often do things that you are not expecting you can use -- 

`\rm --  "$model_name"/oned_model_mod.f90`

`--` means "end of the options". Otherwise if a user decides to give their model a name starting with -, e.g.

`model_name = -fish_pie`

rm will think you are trying to give -fish_pie as an option to rm.

Reply:
Can you only copy the files you need, rather than copying everything and then removing what is not needed? Two reasons to do this:

1. I'd like to have as few `rm`s as possible when 
2. It is possible that more files get added to the template directories, and if so, they will get copied, but not removed by this script.